source/extensions/resource_monitors/fixed_heap/fixed_heap_monitor.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "envoy/config/resource_monitor/fixed_heap/v2alpha/fixed_heap.pb.validate.h"
+#include "envoy/server/resource_monitor.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ResourceMonitors {
+namespace FixedHeapMonitor {
+
+/**
+ * Helper class for getting memory heap stats.
+ */
+class MemoryStatsReader {
+public:
+  MemoryStatsReader() {}
+  virtual ~MemoryStatsReader() {}
+
+  // Memory reserved for the process by the heap.
+  virtual uint64_t reservedHeapBytes();
```

Comment:
The general style in Envoy for this is to declare pure interfaces for everything and then specialize as real vs. mock. I think what you have here is fine though, so no objections, just sharing an idiom that seems fairly pervasive.